### DISCUSSION TITLE
Heat conduction got wrong physical results in Thermal Hydraulics Module

### DISCUSSION THREAD

**Discussion by NorthMagic** [idaholab/moose#27375] [Q&A Modules: Thermal Hydraulics] [answered]
Hello, I'm testing the heat conduction of solid materials in THM. I loaded the mesh file and specified the top/bottom boundaries with a temperature. But the distribution of T_solid seems wrong.

This is the input:
```
[Materials]
  [mat]
    type = ADGenericConstantMaterial
    block = 'grap_block:graphite'
    prop_names = 'density specific_heat thermal_conductivity'
    prop_values = '1780 710 116'
  []
[]

[Components]
  [grap_block]
    type = HeatStructureFromFile3D
    file = mesh/graphite_block_in.e
    position = '0 0 0'
    initial_T = 300
  []
  
  [bottom_bnd]
  	type = HSBoundarySpecifiedTemperature
  	hs = grap_block
  	boundary = grap_block:bottom
  	T = 300
  []
  
  [top_bnd]
  	type = HSBoundarySpecifiedTemperature
  	hs = grap_block
  	boundary = grap_block:top
  	T = 500
  []
[]

[Preconditioning]
  [pc]
    type = SMP
    full = true
  []
[]

[Executioner]
  type = Transient
  scheme = 'bdf2'

  start_time = 0
  dt = 1
  num_steps = 100
  abort_on_solve_fail = true

  solve_type = 'NEWTON'
  line_search = 'basic'
  nl_rel_tol = 1e-7
  nl_abs_tol = 1e-8
  nl_max_its = 10

  l_tol = 1e-3
  l_max_its = 100
[]

[Outputs]
  exodus = true
[]
```

Here is the T_solid distribution:
<img width="478" alt="image" src="https://github.com/idaholab/moose/assets/63328077/e1c4b5cc-a773-44f2-b863-0c2eb1d45c5b">

**Comment by GiudGiud** [comment, marked as answer]
Hello

You are running a transient
```
[Executioner]
  type = Transient
  scheme = 'bdf2'

  start_time = 0
  dt = 1
```
Is 1 s enough time for the heat to propagate?

Guillaume

**Reply by NorthMagic** [reply]
Yes, you're right. Heat needs more time to propagate. In the mentioned input file, boundary conditions for inner and outer surface are not defined. Are the Adiabatic Boundary Condition default setting for undefined boundaries in THM?

**Reply by GiudGiud** [reply]
yes, that arises from how diffusion is implemented with an integration by parts
